fe/fe-core/src/main/java/com/starrocks/sql/optimizer/base/EquivalenceClasses.java

```diff
@@ -37,7 +37,12 @@ public EquivalenceClasses clone() {
         final EquivalenceClasses ec = new EquivalenceClasses();
         for (Map.Entry<ColumnRefOperator, Set<ColumnRefOperator>> entry :
                 this.columnToEquivalenceClass.entrySet()) {
-            ec.columnToEquivalenceClass.put(entry.getKey(), Sets.newLinkedHashSet(entry.getValue()));
+            if (!ec.columnToEquivalenceClass.containsKey(entry.getKey())) {
```

Comment:
It's better to put this into the use of `clone` method? `clone` method should deep copy heree?

Reply:
no, if you use deep copy here, you may add new column into equivalence class, there will be bug